https://bugs.gentoo.org/show_bug.cgi?id=284071


------- Comment #2 From Randall Wald 2009-09-10 08:48:52 0000 [reply] -------

After looking a bit at the code that was breaking, I realized that the problem
occurred because the developers had created modified versions of some system
gstreamer header files (notably, adding the gst_adapter_get_buffer function),
but then used #includes which referred to the system copies, not the new copies
they wanted. I made a patch to switch which copies of these files are included,
and this let my build of schroedinger pass QA. I'll attach my patch and the
modified ebuild which uses it.

------- Comment  #3 From Randall Wald  2009-09-10 08:53:04 0000  [reply] -------

Created an attachment (id=203664) [edit]
Patch to change which headers are included in a few key files

This patch modifies three files (gstbasevideoparse.h, gstbasevideodecoder.h,
and gstbasevideocodec.h) to use copies of gstbasevideocodec.h and
gstbasevideoutils.h included with schroedinger's source tarball, rather than
using the system copies of these files. This is important because the
schroedinger copy of gstbasevideoutils.h defines the function
gst_adapter_get_buffer, while the system copy (at least, on my system) does
not.


diff -ur schroedinger-1.0.7/gst-libs/gst/video/gstbasevideoparse.h.orig schroedinger-1.0.7/gst-libs/gst/video/gstbasevideoparse.h
--- schroedinger-1.0.7/gst-libs/gst/video/gstbasevideoparse.h.orig	2009-09-10 04:37:44.206319157 -0400
+++ schroedinger-1.0.7/gst-libs/gst/video/gstbasevideoparse.h	2009-09-10 04:38:06.719330934 -0400
@@ -20,8 +20,8 @@
 #ifndef _GST_BASE_VIDEO_PARSE_H_
 #define _GST_BASE_VIDEO_PARSE_H_
 
-#include <gst/video/gstbasevideocodec.h>
-#include <gst/video/gstbasevideoutils.h>
+#include "gstbasevideocodec.h"
+#include "gstbasevideoutils.h"
 
 G_BEGIN_DECLS
 
diff -ur schroedinger-1.0.7/gst-libs/gst/video/gstbasevideodecoder.h.orig schroedinger-1.0.7/gst-libs/gst/video/gstbasevideodecoder.h
--- schroedinger-1.0.7/gst-libs/gst/video/gstbasevideodecoder.h.orig	2009-09-10 04:38:21.985071525 -0400
+++ schroedinger-1.0.7/gst-libs/gst/video/gstbasevideodecoder.h	2009-09-10 04:38:49.917081149 -0400
@@ -20,7 +20,7 @@
 #ifndef _GST_BASE_VIDEO_DECODER_H_
 #define _GST_BASE_VIDEO_DECODER_H_
 
-#include <gst/video/gstbasevideocodec.h>
+#include "gstbasevideocodec.h"
 
 G_BEGIN_DECLS
 
diff -ur schroedinger-1.0.7/gst-libs/gst/video/gstbasevideocodec.h.orig schroedinger-1.0.7/gst-libs/gst/video/gstbasevideocodec.h
--- schroedinger-1.0.7/gst-libs/gst/video/gstbasevideocodec.h.orig	2009-09-10 04:39:10.407323883 -0400
+++ schroedinger-1.0.7/gst-libs/gst/video/gstbasevideocodec.h	2009-09-10 04:39:25.443233871 -0400
@@ -22,7 +22,7 @@
 
 #include <gst/gst.h>
 #include <gst/video/gstvideocompat.h>
-#include <gst/video/gstbasevideoutils.h>
+#include "gstbasevideoutils.h"
 
 G_BEGIN_DECLS
 

